src/valid/validate.cc

```diff
@@ -52,7 +65,7 @@ bool BeginCode(ValidCtx& ctx, Location loc) {
   ctx.label_stack.clear();
   ctx.locals.Reset();
   // Don't validate the index, should have already been validated at this point.
-  if (function.type_index < ctx.defined_type_count) {
+  if (function.type_index < ctx.types.size()) {
```

Comment:
Is this needed give then above? I mean aren't the same at this point?

Reply:
Not yet, `BeginCode()` is called first, then `Validate(..., At<DefinedType>)`, then `EndCode()`. So in `BeginCode()`, `ctx.types.size()` and `ctx.defined_type_count` are not necessarily equal. And anyway, I think it's a little clearer to use `ctx.types.size()` here, since we use this index directly below: `ctx.types[function.type_index]`.